test_libs/pyspec/eth2spec/test/helpers/block.py

```diff
@@ -33,26 +37,45 @@ def sign_block(spec, state, block, proposer_index=None):
             domain_type=spec.DOMAIN_RANDAO,
         )
     )
-    block.signature = bls_sign(
-        message_hash=signing_root(block),
+
+
+# Fully ignore the function if BLS is off, beacon-proposer index calculation is slow.
+@only_with_bls()
+def apply_sig(spec, state, signed_block, proposer_index=None):
+    block = signed_block.msg
+
+    proposer_index = get_proposer_index_maybe(spec, state, block.slot, proposer_index)
+    privkey = privkeys[proposer_index]
+
+    signed_block.signature = bls_sign(
+        message_hash=hash_tree_root(block),
         privkey=privkey,
         domain=spec.get_domain(
             state,
             spec.DOMAIN_BEACON_PROPOSER,
             spec.compute_epoch_at_slot(block.slot)))
 
 
+def sign_block(spec, state, block, proposer_index=None):
+    signed_block = spec.SignedBeaconBlock(msg=block)
+    apply_sig(spec, state, signed_block, proposer_index)
+    return signed_block
+
+
+def transition_unsigned_block(spec, state, block):
```

Comment:
why this instead of using `state_trnasition` with the extra checks set to `False`?

Reply:
Different input type. Just the bare `BeaconBlock`, no signature placeholder, and thus explicitly unsigned.